interface/smart/register-app.php

```diff
@@ -225,6 +280,18 @@ function toggleSelectAll(evt) {
         <div class="<?php echo $formarea; ?>">
             <h3 class="card-title text-center"><?php echo xlt("App Registration Form"); ?></h3>
             <div>
+                <div class="form-check form-check-inline">
+                    <input type="radio" class="form-check-input" id="appTypeConfidential" name="appType" value="private" checked="checked"/>
+                    <label for="appTypeConfidential" class="form-check-label pr-2"><?php echo xlt('Confidential'); ?></label>
+                    <input type="radio" class="form-check-input" id="appTypePublic" name="appType" value="public"/>
+                    <label for="appTypePublic" class="form-check-label"><?php echo xlt('Public'); ?></label>
+                </div>
+                <div class="row">
+                    <div class="col alert alert-info">
+                        <p><?php echo xlt("system, user, and offline_access scopes require confidential app permissions."); ?></p>
+                        <p><?php echo xlt("Confidential apps are applications that are able to safely and securely store a secret.  Browser based and many mobile applications do not satisfy this security constraint"); ?></p>
```

Comment:
in translation constants, need to limit to single spacing (remove the double space after the period here). This is because the google doc sometimes truncates this from 2 spaces to 1 and then breaks the constant.

Reply:
Good to know.  I've always been a two space guy, I guess I'll have to learn to adjust :)